sycl/source/detail/platform_impl.cpp

```diff
@@ -49,6 +50,109 @@ platform_impl_host::get_devices(info::device_type dev_type) const {
   return res;
 }
 
+struct DevDescT {
+  const char *devName = nullptr;
+  int devNameSize = 0;
+
+  const char *minDriverVer = nullptr;
+  int minDriverVerSize = 0;
+};
+
+static std::vector<DevDescT> getWhiteListDesc() {
+  // TODO: Replace with const char *str =
+  // SYCLConfig<SYCL_DEVICE_WHITE_LIST>::get();
+  const char *str = getenv("SYCL_DEVICE_WHITE_LIST");
+  if (!str)
+    return {};
+
+  std::vector<DevDescT> decDescs;
+  const char devNameStr[] = "DeviceName";
+  const char driverVerStr[] = "DriverVersion";
+  decDescs.emplace_back();
+  while (*str != '\0') {
+    const char **dstPtr = nullptr;
+    int *Size = nullptr;
+
+    // -1 to avoid comparing null terminator
+    // TODO: Change devNameStr so it doesn't contain null terminator
+    if (0 == strncmp(devNameStr, str, sizeof(devNameStr) - 1)) {
+      dstPtr = &decDescs.back().devName;
+      Size = &decDescs.back().devNameSize;
+      str += sizeof(devNameStr);
+    } else if (0 == strncmp(driverVerStr, str, sizeof(driverVerStr) - 1)) {
+      dstPtr = &decDescs.back().minDriverVer;
+      Size = &decDescs.back().minDriverVerSize;
+      str += sizeof(driverVerStr);
+    }
+
+    if ('\0' == *str || ',' == *str || !dstPtr) {
+      // Malformed white list.
+      // TODO: throw an exception here.
+      break;
+    }
+
+    *dstPtr = str;
+
+    // Skip value
+    while (*str != '\0' && *str != ',' && *str != '|')
+      ++str;
+
+    *Size = str - *dstPtr;
+
+    // | means that the is another filter
+    if(*str == '|')
+      decDescs.emplace_back();
+
+    ++str;
+  }
+
+  return decDescs;
+}
+
+static void filterWhiteList(vector_class<RT::PiDevice> &pi_devices) {
+  const std::vector<DevDescT> whiteList(getWhiteListDesc());
+  if (whiteList.empty())
+    return;
+
+  for (int i = 0; i < (int)pi_devices.size();) {
+    RT::PiDevice dev = pi_devices[i];
+    const string_class devName =
+        sycl::detail::get_device_info<string_class, info::device::name>::_(dev);
+
+    const string_class minDriverVer =
+        sycl::detail::get_device_info<string_class,
+                                      info::device::driver_version>::_(dev);
+
+    bool found = false;
+    for (const DevDescT &desc : whiteList) {
+      // At least device name is required filed to consider the filter
```

Comment:
```suggestion
      // At least device name is required field to consider the filter
```

? Or something like `to be specified` instead of `field`